utils/tfw_logger.cc

```diff
@@ -0,0 +1,461 @@
+
+	promise.set_value();
+}
+catch (...) {
+	promise.set_exception(std::current_exception());
+}
+
+void
+sig_handler([[maybe_unused]] int  sig_num) noexcept
+{
+	stop_flag = true;
+	spdlog::info("Stopping daemon...");
+}
+
+void
+set_sig_handlers()
+{
+	struct sigaction sa;
+
+	sigemptyset(&sa.sa_mask);
+	sigaddset(&sa.sa_mask, SIGHUP);
+	sigaddset(&sa.sa_mask, SIGINT);
+	sigaddset(&sa.sa_mask, SIGTERM);
+
+	sa.sa_handler = sig_handler;
+	sa.sa_flags = SA_RESTART;
+
+	sigaction(SIGHUP, &sa, NULL);
+	sigaction(SIGINT, &sa, NULL);
+	sigaction(SIGTERM, &sa, NULL);
+}
+
+void stop_daemon()
+{
+	pid_t pid;
+	std::ifstream pid_file(pid_file_path);
+
+	std::cout << "Stopping daemon..." << std::endl;
+
+	if (!pid_file)
+		throw Except("No PID file found. Is the daemon running?");
+
+	pid_file >> pid;
+	pid_file.close();
+
+	if (pid <= 0)
+		throw Except("Invalid PID in PID file.");
+
+	if (kill(pid, SIGTERM) < 0)
+		throw Except("Failed to stop daemon: {}", strerror(errno));
+
+	while (1) {
+		if (kill(pid, 0) == -1 && errno == ESRCH)
+			break;
+		std::this_thread::sleep_for(
+			std::chrono::milliseconds(WAIT_FOR_STOP));
+        }
+
+	std::cout << "Daemon stopped." << std::endl;
+}
+
+int
+main(int argc, char* argv[])
+{
+	std::vector<std::thread> thrs;
+	std::vector<std::future<void>> futures;
+	unsigned int i;
+	long int cpu_cnt;
+	int fd = -1, res = 0;
+	bool stop = false;
+
+	try {
+		po::options_description desc{"Usage: tfw_logger "
+					     "([options] <host> <log>) | --stop"};
+		desc.add_options()
+			("help,h", "show this message and exit")
+			("host,H", po::value<std::string>(),
+				"clickserver host address (required)")
+			("log,l", po::value<std::string>(),
+				"log path (required)")
+			("ncpu,n", po::value<unsigned int>(),
+				"manually specifying the number of CPUs")
+			("stop,s", po::bool_switch(&stop), "stop the daemon")
+			("user,u", po::value<std::string>(), "clickhouse user")
+			("password,p", po::value<std::string>(),
+			 "clickhouse password")
+			;
+		po::positional_options_description pos_desc;
+		pos_desc.add("host", 1);
+		pos_desc.add("log", -1);
+		po::variables_map vm;
+		po::store(po::command_line_parser(argc, argv)
+			.options(desc)
+			.positional(pos_desc)
+			.run(),
+			vm);
+		po::notify(vm);
+
+		if (stop) {
+			if (vm.size() > 1)
+				throw Except("--stop can't be used with "
+					     "another arguments");
+			stop_daemon();
+			return 0;
+		}
+
+		if (vm.count("help")) {
+			std::cout << desc << std::endl;
+			return 0;
+		}
+
+		if (!vm.count("host")) {
+			std::cerr << "'host' argument is requred" << std::endl;
+			return 1;
+		}
+
+		if (!vm.count("log"))
+			throw Except("please, specify log path");
+
+		try {
+			auto logger = spdlog::basic_logger_mt("access_logger",
+				vm["log"].as<std::string>());
+			spdlog::set_default_logger(logger);
+		}
+		catch (const spdlog::spdlog_ex &ex) {
+			throw Except("Log init failed: ", ex.what());
+		}
+		spdlog::set_level(spdlog::level::info);
```

Comment:
I have a problem. I don't see any logs in file. Could anyone check it also. Is it only my problem or not?

Reply:
Fixed.